### DISCUSSION TITLE
Sitemaps

### DISCUSSION THREAD

**Discussion by aleczratiu** [vercel/next.js#56905] [Help]
### Summary

Has anyone tried generating multiple sitemaps in nextjs13?

I have to deal with an interesting situation where I already have in an old framework multiple sitemaps created based on some categories (not so important). My concern is if through nextjs is doable to do that or should I add all the slugs as they come directly in one sitemap file?

**Comment by neslob** [comment]
It is announced.

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap

> In the future, we will support multiple sitemaps and sitemap indexes.


Currently I do; waiting for the app router solution.

```
// sitemap.xml.tsx
const sitemaps = [
  'sitemap-blog.xml',
  'sitemap-news.xml',
...
];
...

export const getServerSideProps: GetServerSideProps = async ({ res }) => {
  const sitemap = `<?xml version="1.0" encoding="UTF-8"?>
  <sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">

    ${sitemaps
      .map((url) => {
        return `
      <sitemap>
        <loc>${appendBaseUrl(url)}</loc>
        <lastmod>${new Date().toISOString()}</lastmod>
      </sitemap>
      `;
      })
      .join('')}
  </sitemapindex>
`;

  res.setHeader('Content-Type', 'text/xml');
  res.write(sitemap);
  res.end();

  return {
    props: {},
  };
};

export default Sitemap;
```

**Comment by programming-with-ia** [comment]
A better sitemap generator for Next.js App Router - full support for images, videos, hreflang alternates, sitemap indexes, custom endpoints, and real caching control with drop-in migration. [View on GitHub](https://github.com/programming-with-ia/better-next-sitemap)

**Comment by suryakantamangaraj** [comment]
I’m using the App Router approach: I created separate sitemap routes for each content type and a sitemap index that references them.
Create these route files and have each return its own sitemap:
```
app/sitemap-blog.xml/route.ts
app/sitemap-products.xml/route.ts
app/sitemap-pages.xml/route.ts
```
Then add a sitemap index that points to those sitemaps:
```
app/sitemap.xml/route.ts
```

**Reply by programming-with-ia** [reply]
nice setup 👍. since you are using the app router with separate sitemap routes, you might like [better-next-sitemap - generators](https://github.com/programming-with-ia/better-next-sitemap#3-advanced-generators-full-control)

it auto generates sitemap indexes and named generators, so you do not have to wire everything manually.

if you try it, i would love your feedback.
